src/lib/annotations/doc/DocAnnotator.js

```diff
@@ -501,6 +515,50 @@ class DocAnnotator extends Annotator {
     //--------------------------------------------------------------------------
 
     /**
+     * Handles changes in text selection. Used for mobile highlight creation.
+     *
+     * @return {void}
+     */
+    onSelectionChange(event) {
+        // Do nothing if in a text area
+        if (document.activeElement.nodeName.toLowerCase() === 'textarea') {
+            return;
+        }
+
+        const selection = window.getSelection();
+
+        // If we're creating a new selection, make sure to clear out to avoid
+        // incorrect text being selected
+        if (!this.lastSelection || (this.lastSelection && selection.anchorNode !== this.lastSelection.anchorNode)) {
+            this.highlighter.removeAllHighlights();
+        }
+
+        // Bail if mid highlight and tapping on the screen
+        if (!docAnnotatorUtil.isValidSelection(selection)) {
+            this.lastSelection = null;
+            this.lastHighlightEvent = null;
+            this.createHighlightDialog.hide();
+            return;
+        }
+
+        if (!this.createHighlightDialog.isVisble) {
+            this.createHighlightDialog.show(this.container);
+        }
+
+        // Set all annotations that are in the 'hover' state to 'inactive'
+        Object.keys(this.threads).forEach((threadPage) => {
+            this.getHighlightThreadsOnPage(threadPage).filter(isThreadInHoverState).forEach((thread) => {
```

Comment:
I think there was another instance where you couldn't use reset() but seems like docHighlightThread.reset() just does the same thing as what you've done below?

https://github.com/box/box-content-preview/blob/master/src/lib/annotations/doc/DocHighlightThread.js#L80